DOKS-automatic-node-repair/README.md

```diff
@@ -1 +1,402 @@
+DOKS_DEBUG_POD_NAME=$(kubectl get pods -l name=doks-debug -ojsonpath='{.items[0].metadata.name}' -n kube-system)
+kubectl exec -it "$DOKS_DEBUG_POD_NAME" -n kube-system -- chroot /host systemctl stop kubelet
+```
+
+## Observation and Results
+
+After the `kubelet` stop command is issued you'll be kicked out of the `kubectl exec` session which is a good sign because this means that the `Node Controller` lost its connection with the affected node where the `kubelet` was killed.
+ 
+The results can be seen in the other terminal window where the `watch` was set (notice the `NotReady` state of the affected node):
+
+```
+NAME            STATUS     ROLES    AGE    VERSION
+basicnp-8hc5d   Ready      <none>   28h    v1.21.2
+basicnp-8hx1a   NotReady   <none>   144m   v1.21.2
+```
+
+After the `DELAY_NODE_CREATION` time interval expired, the node should vanish which is the expected behavior:
+
+```
+NAME            STATUS   ROLES    AGE   VERSION
+basicnp-8hc5d   Ready    <none>   28h   v1.21.2
+```
+
+Let's see what `Digital Mobius` has to say about this situation: will it notice and take the appropriate action? Open a terminal window and inspect the logs first:
+
+```bash
+kubectl logs -l app.kubernetes.io/name=digital-mobius -n maintenance
+```
+
+The output should look like below (watch for the `Recycling node {...}` lines):
+
+```
+     _ _       _ _        _                      _     _           
+  __| (_) __ _(_) |_ __ _| |     _ __ ___   ___ | |__ (_)_   _ ___ 
+ / _` | |/ _` | | __/ _` | |    | '_ ` _ \ / _ \| '_ \| | | | / __|
+| (_| | | (_| | | || (_| | |    | | | | | | (_) | |_) | | |_| \__ \
+ \__,_|_|\__, |_|\__\__,_|_|    |_| |_| |_|\___/|_.__/|_|\__,_|___/
+         |___/                                                     
+time="2021-08-06T08:29:52Z" level=info msg="Starting Digital Mobius 0.1.4 \n"
+time="2021-08-06T11:13:09Z" level=info msg="Recyling node {11bdd0f1-8bd0-42dc-a3af-7a83bc319295 f8d76723-2b0e-474d-9465-d9da7817a639 379826e4-8d1b-4ba4-97dd-739bbfa69023}"
+...
+```
+In the terminal window where the `watch` was set for `kubectl get nodes` a new node should appear after a minute or so, replacing the old one (notice that it has a different `ID` and a new `AGE` value):
+
+```
+NAME            STATUS   ROLES    AGE   VERSION
+basicnp-8hc5d   Ready    <none>   28h   v1.21.2
+basicnp-8hoav   Ready    <none>   22s   v1.21.2
+```
+
+**The experiment was a success! The node was recycled.**
+
+## Final Notes
+
+In the end we can see that it's very easy to handle and overcome some special situations if having the right tools at hand. In this tutorial we talked about a simple solution that can automatically recover cluster nodes in case the `kubelet` dies or becomes unresponsive. This can happen when the worker node is overloaded  or due to some unexpected networking issues. High load can be caused by not following good practices when it comes to `Pod resources limits`, like not setting them at all or using inadequate values.
```

Comment:
```suggestion
In this tutorial, we automatically recover cluster nodes in case the kubelet dies or becomes unresponsive. This can happen when the worker node is overloaded  or due to some unexpected networking issues. High load can be caused by not following good practices when it comes to `Pod resources limits`, like not setting them at all or using inadequate values.
```